pulp_ansible/app/tasks/collectionversion_index.py

```diff
@@ -154,9 +157,14 @@ def update_index(distribution=None, repository=None, repository_version=None, is
     deprecations = AnsibleCollectionDeprecated.objects.filter(pk__in=deprecations)
     deprecations_set = {(x.namespace, x.name) for x in deprecations}
 
-    # find all namespaces in the repo version
-    namespaces = repository_version.get_content(content_qs=AnsibleNamespaceMetadata.objects).all()
-    namespaces = {x.name: x for x in namespaces}
+    # find all the most recent namespace metadata in the repo version
+    namespaces = {}
+    for ns in AnsibleNamespaceMetadata.objects.all():
```

Comment:
this looks like it gets all `AnsibleNamespaceMetadata` objects, not just the ones scoped to the repository_version

Reply:
fixed